### DISCUSSION TITLE
Plugin alternative in 3.0

### DISCUSSION THREAD

**Discussion by NinjaDevRN** [jpudysz/react-native-unistyles#274] [General]
Hi, thank you for your library first of all!

In the documentation, we have an example about layout scaling with plugin usage:

> Scale your UI automatically based on screen size. It can replace [react-native-responsive-screen](https://github.com/marudy/react-native-responsive-screen#readme).

How should we adapt our code to be ready for version 3.0 of the library?

**Comment by jpudysz** [comment]
Scale your layout based on your function in theme. Plugins are just too slow for C++. I parse stylesheet natively in less than a 1ms. I just don't want to map it again in JavaScript. 

Anything that was possible with plugins can be done with theme and functions.

**Reply by NinjaDevRN** [reply]
Thank you 🙏
